### PR TITLE
Fix scope.$apply to be safe

### PR DESCRIPTION
**Pull request by blittle**
https://sentry.canopytax.com/canopy/frontend-production-1/issues/24864/events/616896/

### CODE REVIEW

File: src/multi-selector.js [hunk 1]
```diff
@@ -248,3 +246,13 @@ angular.module('cp-multi-selector')
 			}
 		}
 	}]);
+
+	function safeApply(scope, fn) {
```
Comment (blittle) [inline src/multi-selector.js:250]: https://coderwall.com/p/ngisma/safe-apply-in-angular-js

File: src/multi-selector.js [hunk 2]
```diff
@@ -196,7 +196,7 @@ angular.module('cp-multi-selector')
 				 */
 				$('body').on('click.cpmultiselector' + id, function(e) {
 					if(!$(e.target).closest(el).length) {
-						scope.$apply(function() {
```
Comment (blittle) [inline src/multi-selector.js:199]: This is the root of the cause. You can't call `scope.$apply` arbitrarily in a dom event handler. A digest might already be underway.

File: src/multi-selector.js (hunk 1 above)
Comment (blittle) [reply src/multi-selector.js:250]: Kinda a hack to test if a digest is already in progress. If it is, don't `$apply`

**Review by keithhalterman — APPROVED** [review]
🍃  Cool hack detecting digests in progress

**Pull request merged**